lib/Versions/VersionsBackend.php

```diff
@@ -275,27 +281,27 @@ public function getAllVersionedFiles(FolderDefinitionWithMappings $folder): arra
 		return array_combine($fileIds, $files);
 	}
 
-	public function deleteAllVersionsForFile(int $folderId, int $fileId): void {
-		$versionsFolder = $this->getVersionsFolder($folderId);
+	public function deleteAllVersionsForFile(FolderDefinition $folder, int $fileId): void {
+		$versionsFolder = $this->getVersionsFolder($folder);
 		try {
 			$versionsFolder->get((string)$fileId)->delete();
 			$this->groupVersionsMapper->deleteAllVersionsForFileId($fileId);
 		} catch (NotFoundException) {
 		}
 	}
 
-	private function getVersionsFolder(int $folderId): Folder {
-		try {
-			/** @var Folder $folder */
-			$folder = $this->appFolder->get('versions/' . $folderId);
-
-			return $folder;
-		} catch (NotFoundException) {
-			/** @var Folder $trashRoot */
-			$trashRoot = $this->appFolder->nodeExists('versions') ? $this->appFolder->get('versions') : $this->appFolder->newFolder('versions');
-
-			return $trashRoot->newFolder((string)$folderId);
+	public function getVersionsFolder(FolderDefinition $folder): Folder {
+		$mountPoint = '/dummy/files_versions/groupfolders/' . $folder->id;
+		$mount = $this->mountManager->find($mountPoint);
+		if ($mount->getMountPoint() !== $mountPoint) {
+			$trashMount = $this->mountProvider->getVersionsMount(
+				$folder,
+				$mountPoint,
+				$this->storageFactory,
+			);
+			$this->mountManager->addMount($trashMount);
 		}
```

Comment:
Is this named trashMount by copy/paste as well?
I’d welcome a comment here, because I do not understand what `$mount->getMountPoint() !== $mountPoint` implies.

Reply:
Also NotFoundException from calling `find` should be caught, no?

Reply:
Added a comment,

`find` throwing a `NotFoundException` means that the root storage isn't setup so things are very broken and the error should be propagated.